packages/pg/lib/stream.js

```diff
@@ -18,8 +18,8 @@ module.exports.getStream = function getStream(ssl) {
  * @returns {Duplex}
  */
 module.exports.getSecureStream = function getSecureStream(options) {
-  var tls = require('tls')
-  if (tls.connect) {
+  if (process.release && process.release.name === 'node') {
+    var tls = require('tls')
```

Comment:
nit: if you want extreme pedantry (I mean...who doesn't, right? 😉 ) putting the conditional (`process.release && process.release.name === 'node') into a predicate function `isNode()` or something would DRY this up a bit.   I'm also slightly concerned how this impacts bun/deno? Is there any way to invert the check and see if you're inside cloudflare instead?  Then if you're not in cloudflare but you're in a fully node compatible environment (be that node, deno, bun) we're using the node side of things.  Obviously this is subject to change as the node/deno/bun/cf/browser/wasm environments continue to converge...just my thoughts here.

Reply:
> Is there any way to invert the check and see if you're inside cloudflare instead?

Yes, that's what I meant by checking if you're in CF (as opposed to checking if you're not in node).

I'm not sure if they expose anything like that. If not, they definitely should add a `process.env.CLOUDFLARE_WEB_WORKER` env similar to how GitHub does for actions.

> Then if you're not in cloudflare but you're in a fully node compatible environment (be that node, deno, bun) we're using the node side of things.

Reminds me of the io.js days!

Reply:
😢

Reply:
Of course trying to import from `cloudflare:sockets` has exactly this behaviour. It is checking whether we are in a Worker or not.

Reply:
The import of `pg-cloudflare` outside of a worker is an empty package.

Reply:
I also considered a "Cloudflare specific global" such as `HtmlRewriter` which doesn't exist in node or deno... but it does exist in bun!

I am asking internally to see if there is a more reliable way to determine the runtime.